api/get.js

```diff
@@ -0,0 +1,179 @@
+                    if (err) {
+                        reject(err);
+                        return;
+                    }
+                    equals(req.params, data)
+                        .then(function() {
+                            console.log('File', path.basename(file), 'found!');
+                            resolve(file);
+                        })
+                        .fail(function() {
+                            reject(new Error('Not found!'));
+                        });
+                });
+            });
+        });
+        return Q.allSettled(promises)
+            .then(function(result) {
+                var chain = _.chain(result)
+                    .filter(function(file) { return file.state === 'fulfilled'; })
+                    .first();
+                result = chain.value().value;
+                console.log('found:', path.basename(result));
+                return Q.Promise(function(resolve, reject) {
+                    //console.log('response found... reading files...');
+                    Q.allSettled([
+                        Q.nfcall(fs.readFile, result.replace(/\.req$/, '.stats'), { encoding: 'utf8' }),
+                        Q.nfcall(fs.readFile, result.replace(/\.req$/, '.res'), { encoding: 'utf8' })
+                    ])
+                    .then(function(files) {
+                        var stats = JSON.parse(files[0].value);
+                        res.status(stats.status);
+                        _.each(stats.headers, function(value, key) {
+                            res.setHeader(key, value);
+                        });
+                        res.write(files[1].value);
+                        res.end();
+                        resolve(result);
+                    })
+                    .fail(reject);
+                });
+            });
+    },
+    
+    request: function(url, fileList, req, res, next) {
+        return Q.Promise(function(resolve, reject) {
+            console.log('file not found, requesting from', url + req.url);
+            request(url + req.url, function(error, response, body) {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+                if (response.statusCode === 200) {
+                    this.save(req.params, response, body, fileList.length)
+                        .then(function() {
+                            res.status(response.statusCode);
+                            _.each(response.headers, function(value, key) {
+                                res.setHeader(key, value);
+                            });
+                            res.write(body);
+                            res.end();
+                            resolve();
+                        })
+                        .fail(reject);
+                    return;
+                }
+                reject(new Error(body));
+            }.bind(this));
+        }.bind(this));
```

Comment:
.bind(this) are all over the place, it's becoming a bit unwieldy, is there a better way?